### PR TITLE
Feat/191 add calculate embodied emissions

### PR DESCRIPTION
**Pull request by GuillaumeBarra**

### CODE REVIEW

File: src/view/popup/__tests__/calculateEmbodiedEmissions.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const mockDevice: Map<CountryName, number> = new Map([["iPhone 11", 1]]);
+
+const mockDevices: Map<CountryName, number> = new Map([
+    ["iPhone 11", 0.2],
+    ["iPhone 7", 0.2],
+    ["Google Pixel 5", 0.4],
+]);
+
+describe("calculateCarbon", () => {
```
Comment (juskek) [inline src/view/popup/__tests__/calculateEmbodiedEmissions.test.ts:12]:
```suggestion
describe("calculateEmbodiedEmissions", () => {
```

File: src/view/popup/__tests__/calculateEmbodiedEmissions.test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    it("country should return corresponding value", () => {
```
Comment (juskek) [inline src/view/popup/__tests__/calculateEmbodiedEmissions.test.ts:13]: can we replace country and countries to device and devices in this file please? :)

File: src/view/popup/utils/calculateEmbodiedEmissions.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {
+    AVERAGE_DEVICE_LIFETIME_CO2_EMISSIONS_GRAMS,
+    DeviceName,
+    DEVICE_LIFETIME_CO2_EMISSIONS_GRAMS
+} from "../../../data/constants/DeviceEmissions";
+
+const AVG_DEVICE_LIFETIME = (4 * 365 + 1) * 24 * 3600;
+
+export const calculateEmbodiedEmissions = (
+    flowLength: number,
```
Comment (juskek) [inline src/view/popup/utils/calculateEmbodiedEmissions.ts:10]: Should we add units to flowLength? e.g. flowLengthMs or flowLengthSeconds

**Review by juskek — CHANGES_REQUESTED** [review]

File: src/view/popup/utils/calculateEmbodiedEmissions.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import {
+    AVERAGE_DEVICE_LIFETIME_CO2_EMISSIONS_GRAMS,
+    DeviceName,
+    DEVICE_LIFETIME_CO2_EMISSIONS_GRAMS
+} from "../../../data/constants/DeviceEmissions";
+
+const AVG_DEVICE_LIFETIME = (4 * 365 + 1) * 24 * 3600;
```
[the remark this answers is not in the document]
Comment (GuillaumeBarra) [reply src/view/popup/utils/calculateEmbodiedEmissions.ts:7]: 365 days per year, except once every 4 years we get an extra day

[the remark this answers is not in the document]
Comment (juskek) [reply src/view/popup/utils/calculateEmbodiedEmissions.ts]: should it be `AVG_DEVICE_LIFETIME_YEARS*365.25` then? 
if not when someone changes `const AVG_DEVICE_LIFETIME_YEARS = 4;` then it won't make sense

[the remark this answers is not in the document]
Comment (GuillaumeBarra) [reply src/view/popup/utils/calculateEmbodiedEmissions.ts]: yeah good catch

**Review by juskek — APPROVED** [review]

**Pull request merged**